src/main/webapp/app/guided-tour/guided-tour-step.model.ts

```diff
@@ -25,6 +26,10 @@ export abstract class TourStep {
     userInteractionEvent?: UserInteractionEvent;
     /** Skips this step if the selector is not found, else the setStepAlreadyFinishedHint will be called by the guided tour service */
     skipStepIfNoSelector?: boolean;
+    /** List of tasks that have to be completed */
+    modelingTask?: GuidedTourModelingTask;
```

Comment:
why does tour step need to be connected to modeling task? the documentation is also misleading, this is not a list.
Can we not handle this differently or more general?

There is also no `programmingTask`

Reply:
@krusche I have created an issue for that and will implement it in a follow-up PR: https://github.com/ls1intum/Artemis/issues/1062

Reply:
@krusche I changed that as discussed and created a `ModelingTaskTourStep` for that matter